.travis.yml

```diff
@@ -113,6 +113,9 @@ matrix:
             # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
             - flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
           script: true
+        - os: linux
+          compiler: gcc
+          env: CONFIGURE_TARGET="linux-generic32" CONFIG_OPTS="--strict-warnings no-shared no-dso no-pic no-autoload-config no-engine no-comp no-cms no-dh no-dsa no-ec2m no-srp no-des no-mdc2 no-md4 no-rc2 no-rc4 no-bf no-camellia no-cast no-idea no-rmd160 no-seed no-whirlpool no-ocsp no-ocb no-sm2 no-sm3 no-sm4 no-ssl3 no-ssl3-method no-psk no-heartbeats no-nextprotoneg no-sctp no-srtp no-ct no-async no-asm -DOPENSSL_SMALL_FOOTPRINT"
```

Comment:
no-heartbeats is a deprecated option so probably doesn't need to be there

Reply:
is this completely ignored, then yes.
I would like to add no-securememory once that PR gets merged or
-DOPENSSL_NO_SECURE_MEMORY instead.

Reply:
It's ignored (there are no instances of OPENSSL_NO_HEARTBEATS in the code).

Reply:
okay, removed no-heartbeats then.

Reply:
I just cross checked against the list of Configurable options and there are quite a few that you don't have here. Not sure if that is deliberate or not. Some perhaps should be included because they would disable large parts of the library and hide some issues that might otherwise be shown. The following might be worthy of consideration:

aria
blake2
chacha
cmac
cmp
dgram
dtls
ecdh
ecdsa
filenames
fips
gost
ktls
md2
multiblock
poly1305
psk
rc5
scrypt
siphash
siv
sse2
threads
ts
ui-console

I would also drop no-sctp since that is the default anyway.

Reply:
This is build spec is derived from a minimal build environment, we use in one of our
embedded products at Softing.  So the target is is mostly reducing footprint of libcrypto
to a minimum, while retaining basic RSA, EC and AES, algorithms.  So yes I think I will
definitely evaluate these options, and I am curious if this allows some improvements
that I have overlooked previously,  so many thanks for this list.

Reply:
I'll remove no-sctp as suggested, (I think the default-disabled options
should be enabled in another build)
no-cmp is already used, no-ecdh and no-ecdsa do not have any effect, except in
removing some tests.
no-fips has no effect on libcrypto / libssl size.
no-md2 is disabled by default, right?
no-ktls probably should add this is disabled on my machine,
however I am not sure, the config script does something stupid:
look for `/usr/$config{cross_compile_prefix}`
IF I WOULD be cross compiling, I WOULD NEVER HAVE MY SYSROOT THERE.
no-psk does not seem to have any effect, at least no on libcrypto/libssl size.
no-rc5 is disabled by default.
no-sse2 is implied by no-asm.
no-threads, saves very few bytes, but is very dangerous, I will not add this one.
the rest should be fine.